ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,359 @@
+                self.output_path.unlink()
+            else:
+                self.log.critical(
+                    f"Output file {self.output_path} exists, "
+                    "use `--overwrite` to overwrite"
+                )
+                sys.exit(1)
+
+        PROV.add_output_file(str(self.output_path))
+
+        if self.skip_parameters is True and self.skip_images is True:
+            self.log.warning("Skip-parameters and skip-images are both" "set to True")
+
+        # Get input Files
+        args = self.parser.parse_args(self.extra_args)
+        self.input_files = args.input_files
+        if self.input_dir is not None:
+            self.input_files.extend(sorted(self.input_dir.glob(self.file_pattern)))
+        if not self.input_files:
+            self.input_files = sorted(Path(".").glob(self.file_pattern))
+
+        # create output file with subarray from first file
+        self.first_subarray = SubarrayDescription.from_hdf(self.input_files[0])
+        self.first_subarray.to_hdf(self.output_path)
+        self.output_file = tables.open_file(self.output_path, mode="a")
+
+    def check_file(self, file):
+        # Check that the file is not broken or any node is missing
+        file_path = file.root._v_file.filename
+        current_subarray = SubarrayDescription.from_hdf(file_path)
+        # Check subarrays
+        if self.first_subarray != current_subarray:
+            self.log.warning(f"Subarray does not match for {file_path}. " "Skip File.")
+            self.skip_file = True
+
+        # Gives warning if tables for listed nodes in 'optional_nodes'
+        # are missing but continues merging the rest of the file. If any
+        # other node from 'required_nodes' is missing, except the given combinations
+        # of traits and blocklist, the whole file will be skipped.
+        for node in required_nodes:
+            if self.is_simu is False and node in allowlist_simu:
+                continue
+            if self.skip_simu_images is True and node is blocklist_images[0]:
+                continue
+            if self.skip_images is True and node in blocklist_images:
+                continue
+            if self.skip_parameters is True and node in blocklist_parameters:
+                continue
+            if node in optional_nodes and node not in file:
+                self.log.warning(
+                    f"{node} is not in {file_path}. Continue with "
+                    "merging file. This table will be incomplete "
+                    "or empty"
+                )
+                continue
+            if node not in file:
+                self.log.warning(f"{node} is not in {file_path}. Skip file")
+                self.skip_file = True
+
+    def add_image_statistics(self, file):
+        # Creates table for image statistics and adds the entries together.
+        # This does not append rows to the existing table
+        image_statistics_path = allowlist_service[0]
```

Comment:
This is not good. If someone reorders allowlist (for whatever reason), this will break. Please just give the needed path here.